spotfinder/spotfinder.cu

```diff
@@ -9,118 +9,278 @@
+          d_dispersion_mask.pitch,   // Output dispersion mask pitch
+          width,                     // Image width
+          height,                    // Image height
+          max_valid_pixel_value,     // Maximum valid pixel value
+          first_pass_kernel_radius,  // Kernel radius
+          first_pass_kernel_radius,  // Kernel radius
+          min_count,                 // Minimum count
+          n_sig_b,                   // Background significance level
+          n_sig_s                    // Signal significance level
+        );
+        cudaStreamSynchronize(
+          stream);  // Synchronize the CUDA stream to ensure the first pass is complete
+
+        if (do_writeout) {
+            printf("First pass complete\n");
+            debug_writeout(
+              d_dispersion_mask.get(),
+              d_dispersion_mask.pitch_bytes(),
+              width,
+              height,
+              stream,
+              "first_pass_dispersion_result",
+              [](uint8_t pixel) { return pixel == MASKED_PIXEL ? 0 : 255; },
+              [](uint8_t pixel) { return pixel != 0; });
+        }
+    }
+
+    /*
+     * Erosion pass ✂
+     * Erode the first pass results.
+     * The surviving pixels are then used as a mask to exclude them
+     * from the background calculation in the second pass.
+    */
+    PitchedMalloc<uint8_t> d_erosion_mask(width, height);
+
+    {  // Scope for erosion pass launch parameters
+        // dim3 threads_per_erosion_block(32, 32);
+        // dim3 erosion_blocks(
+        //   (width + threads_per_erosion_block.x - 1) / threads_per_erosion_block.x,
+        //   (height + threads_per_erosion_block.y - 1) / threads_per_erosion_block.y);
+
+        // Calculate the shared memory size for the erosion kernel
+        // size_t erosion_shared_memory =
+        //   (threads_per_erosion_block.x + 2 * first_pass_kernel_radius)
+        //   * (threads_per_erosion_block.y + 2 * first_pass_kernel_radius)
+        //   * sizeof(uint8_t);
+
+        // Perform erosion
+        erosion_kernel<<<blocks, threads, shared_memory, stream>>>(
+          d_dispersion_mask.get(),
+          d_erosion_mask.get(),
+          mask.get(),
+          d_dispersion_mask.pitch,
+          d_erosion_mask.pitch,
+          mask.pitch,
+          width,
+          height,
+          first_pass_kernel_radius);
+        cudaStreamSynchronize(
+          stream);  // Synchronize the CUDA stream to ensure the erosion pass is complete
+
+        if (do_writeout) {
+            printf("Erosion pass complete\n");
+            debug_writeout(
+              d_erosion_mask.get(),
+              d_erosion_mask.pitch_bytes(),
+              width,
+              height,
+              stream,
+              "eroded_dispersion_result",
+              [](uint8_t pixel) { return pixel == MASKED_PIXEL ? 0 : 255; },
+              [](uint8_t pixel) { return pixel == MASKED_PIXEL; });
```

Comment:
This anonymous scope appears to not be doing anything now?

Reply:
Apologies, I meant to remove these when I mentioned them yesterday and forgot 😓

https://github.com/DiamondLightSource/fast-feedback-service/commit/e57c8e138c0e1f6edc3d3f35b2df9881bff78895